solver/types.go

```diff
@@ -146,6 +146,10 @@ type Op interface {
 
 	// Exec runs an operation given results from previous operations.
 	Exec(ctx context.Context, g session.Group, inputs []Result) (outputs []Result, err error)
+
+	// CountsAsParallelism specifies whether the `Op` should be subject to the parallelism
+	// semaphore.
+	CountsAsParallelism() bool
```

Comment:
Created follow-up proposal that uses `Aquire()` method here #2108 in order to keep resource controls ops specific, not scheduler specific. If you agree with that reasoning, can you change this signature already.

Reply:
:+1: